packages/affiliates/libs/contracts.js

```diff
@@ -0,0 +1,37 @@
+const ethers = require("ethers");
+const assert = require("assert");
+function DecodeLog(abi, meta = {}) {
+  assert(abi, "requries abi");
+  const iface = new ethers.utils.Interface(abi);
+  return (log, props = {}) => {
+    return {
+      ...iface.parseLog(log),
+      ...meta,
+      ...props
+    };
+  };
+}
+function DecodeTransaction(abi, meta = {}) {
+  assert(abi, "requries abi");
+  const iface = new ethers.utils.Interface(abi);
+  return (transaction, props = {}) => {
+    return {
+      ...iface.parseTransaction({ data: transaction.input }),
+      ...meta,
+      ...props
+    };
+  };
+}
+
+function decodeAttribution(
```

Comment:
Definiltey gonna need some comments in this file to understand what's going on haha